cloudahoywriter.lua

```diff
@@ -375,6 +409,42 @@ function maybe_write_data()
     lastWriteTime = os.clock()
 end
 
+-- Called from show_ui() on every draw. Keep this fast.
+-- Decides whether to automatically start or stop recording.
+function automatic_recording_state_check()
+    if CAWR_isPaused == 1 then return end
+
+    if not CAWR_groundSpeed then
+        return -- Possible race with registering datarefs
+    end
+
+    local currentTime = os.time()
+    if userRecordingStateChangeTime
+        and currentTime < userRecordingStateChangeTime + AUTO_RECORDING_DISABLE_SECS then
+        -- Disable automatic state changes due to recent user action.
+        return
+    end
+    if is_recording() then
+        if CAWR_groundSpeed < AUTO_RECORDING_GROUND_SPEED_MPS then
+            autoStopLowSpeedTime = autoStopLowSpeedTime
```

Comment:
If I understand this correctly, it will start the "countdown" when the groundspeed drops below the required groundspeed, but it will maintain its decision to stop recording, even if later the ground speed increases (eg. in the case of a stop and go). Is that what we want?

Reply:
Oh, sorry, I re-read the code, the first if will never be entered so the behavior I described won't happen.
But autoStopLowSpeedTime will remain not-nil and the next time the GS drops, if enough time has elapsed, we'll instantly stop recording?

Reply:
Good catch. Added code to set `autoStopLowSpeedTime = nil` when the groundspeed exceeds the threshold.